File: ch02/python/2.2.1.fib.py
def fib(n):

    a, b = 1, 1

    for i in range(1, n):
        a, b = b, a + b
        print(f'i: {i} -- a: {a}, b: {b}')

    return a

if __name__ == '__main__':
    n = 10
    print(fib(n))

File: ch02/nodejs/2.2.1.fib.js
const fib = n => {
    let a = 1;
    let b = 1;

    for (let i = 1; i < n; i++) {
        let nextNumber = a + b;
        a = b;
        b = nextNumber;
    }

    return a;
}

const n = 10;
console.log(fib(n));
